src/os/bluestore/ZonedFreelistManager.cc

```diff
@@ -272,6 +272,15 @@ std::vector<zone_state_t> ZonedFreelistManager::zoned_get_zone_states(
   return zone_states;
 }
 
+void ZonedFreelistManager::zoned_mark_zone_clean(
```

Comment:
IMO a method getting a reference to a transaction as a parameter would be more universal. One can use it for both piggy-backing an existing transaction or build a new one from scratch

Reply:
Done.

Reply:
IMO this is a proper way indeed. You might want to use submit_transaction_sync() instead though... Please share the assertion if this still doesn't work...

Reply:
Okay, I will try it again and share the assertion.